### ISSUE TITLE
Rule proposal: Require comparator function for `Array#sort()` 

### ISSUE THREAD

**Issue by bmish** [sindresorhus/eslint-plugin-unicorn#2014]
### Description

@typescript-eslint/eslint-plugin has an existing rule [typescript-eslint/require-array-sort-compare](https://typescript-eslint.io/rules/require-array-sort-compare/). See that rule doc for why omitting a comparator leads to common bugs (in both JavaScript and TypeScript).

Because TypeScript provides type information to the rule, the typescript-eslint rule can reliably operate only on actual arrays. But that also means anyone using plain JavaScript code (i.e. without TypeScript/typescript-eslint) cannot benefit from the rule.

We could create a version of this rule that does not depend on type information targeted at JavaScript users. It could simply look for calls like `myArray.sort()`, similar to many of our existing rules like [unicorn/require-array-join-separator](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/require-array-join-separator.md).

If we run into unacceptable high false positives, we could try to add heuristics for detecting when a variable is likely to be or not to be an array (e.g. ignoring classes/objects that have a `sort()` function but are known to not be an array).

We could name this rule `require-array-sort-compare`.

Related:
* #1473

### Fail

```js
array.sort();
```

### Pass

```js
array.sort((a, b) => a - b);
array.sort((a, b) => a.localeCompare(b));
```
